pkg/controller/s3/bucket/lifecycleConfig.go

```diff
@@ -193,3 +190,147 @@ func sortFilterTags(rules []awss3.LifecycleRule) {
 		}
 	}
 }
+
+// LateInitialize does nothing because LifecycleConfiguration might have been be
+// deleted by the user.
+func (in *LifecycleConfigurationClient) LateInitialize(ctx context.Context, bucket *v1beta1.Bucket) error {
+	external, err := in.client.GetBucketLifecycleConfigurationRequest(&awss3.GetBucketLifecycleConfigurationInput{Bucket: awsclient.String(meta.GetExternalName(bucket))}).Send(ctx)
+	if err != nil {
+		// Short stop method for requests without a lifecycle configuration
+		if s3.LifecycleConfigurationNotFound(err) {
+			return nil
+		}
+		return awsclient.Wrap(err, lifecycleGetFailed)
+	}
+
+	// We need the second check here because by default the lifecycle is not set
+	if external.GetBucketLifecycleConfigurationOutput == nil || len(external.Rules) == 0 {
+		return nil
+	}
+
+	in.logger.Debug(fmt.Sprintf("called LateInitialize for %s", reflect.TypeOf(in).Elem().Name()))
+
+	if bucket.Spec.ForProvider.LifecycleConfiguration == nil {
+		bucket.Spec.ForProvider.LifecycleConfiguration = &v1beta1.BucketLifecycleConfiguration{}
+	}
+
+	createLifecycleRulesFromExternal(external.Rules, bucket.Spec.ForProvider.LifecycleConfiguration)
```

Comment:
Unless there is a unique identifier in the type, `id` seems to exists but it's optional, we can late-init only if the array in spec is empty because otherwise, it's not possible to know which element corresponds to which in cloud and whether we should delete or late-init. So, I'd suggest having a simple `if len(bucket.Spec.ForProvider.LifecycleConfiguration.Rules) == 0` condition here and calling a raw creation function similar to `GenerateCORSRule`.